scripts/clean.mjs

```diff
@@ -0,0 +1,308 @@
+      const filePath = path.join(currentDirPath, name)
+      const stat = fs.statSync(filePath)
+
+      if (stat.isDirectory()) {
+        if (!deleteDirectories || !patternCheck(filePath)) {
+          walkSync(filePath)
+          return
+        }
+        if (isGitTracked(filePath)) {
+          log(`Skipping git-tracked directory: ${filePath}`)
+          return
+        }
+
+        if (dry(`Would delete directory: ${filePath}`)) return
+        try {
+          log(`Deleting directory now: ${filePath}`)
+          fs.rmSync(filePath, { recursive: true, force: true })
+        } catch (err) {
+          log(`Failed to delete directory: ${filePath}`, err)
+        }
+        return
+      }
+
+      if (!stat.isFile()) return
+
+      if (!patternCheck(filePath)) return
+
+      if (isGitTracked(filePath)) {
+        log(`Skipping git-tracked file: ${filePath}`)
+        return
+      }
+
+      if (!dry(`Would delete: ${filePath}`)) {
+        try {
+          log(`Deleting now: ${filePath}`)
+          fs.unlinkSync(filePath)
+        } catch (err) {
+          log(`Failed to delete file: ${filePath}`, err)
+        }
+      }
+    })
+  }
+
+  if (deleteDirectories && patternCheck(baseDir)) {
+    if (!dry(`Would delete directory: ${baseDir}`)) {
+      try {
+        log(`Deleting directory now: ${baseDir}`)
+        fs.rmSync(baseDir, { recursive: true, force: true })
+      } catch (err) {
+        log(`Failed to delete directory: ${baseDir}`, err)
+      }
+    }
+    return
+  }
+
+  walkSync(baseDir)
+}
+
+function cleanGenerated() {
+  config.SEARCH_DIRECTORIES.forEach((baseDir) => {
+    if (fs.existsSync(baseDir)) {
+      findAndDelete(baseDir, (filePath) =>
+        config.GENERATED_PATTERNS.some((regex) => regex.test(filePath)),
+      )
+      config.DIRS_TO_DELETE.forEach((dirToDelete) => {
+        findAndDelete(
+          baseDir,
+          (filePath) => filePath.includes(dirToDelete),
+          true, // Indicate that we are deleting directories
+        )
+      })
+    }
+  })
+}
+
+function cleanCaches() {
+  const yarnCacheFolder = execSync('yarn config get cacheFolder', {
+    encoding: 'utf-8',
+  }).trim()
+
+  const cachesToDelete = [...config.CLEAN_CACHES_LIST, yarnCacheFolder]
+
+  cachesToDelete.forEach((item) => {
+    if (fs.existsSync(item)) {
+      findAndDelete(
+        item,
+        (_) => true, // No specific pattern, delete all items in cache
+        true, // Indicate that we are deleting directories
+      )
+    } else {
+      log(`Skipping ${item}: directory does not exist`)
+    }
+  })
+}
+
+function cleanDist() {
+  const baseDir = './'
+  if (fs.existsSync(baseDir)) {
+    findAndDelete(
+      baseDir,
+      (filePath) => path.basename(filePath) === 'dist',
+      true, // Indicate that we are deleting directories
+    )
+  }
+}
+
+function cleanYarn() {
+  if (fs.existsSync('.yarn')) {
+    const r = new RegExp(`\\.yarn/${config.CLEAN_YARN_IGNORES_LIST.join('|')}`)
```

Comment:
Does this not require a `[]` selector around the join statement?

`\\.yarn/[${config.CLEAN_YARN_IGNORES_LIST.join('|')}]`

so that it looks like this: `\\.yarn/[a|b|c]`

rather than: `\\.yarn/a|b|c`

Reply:
Yes, indeed. This would've matched `.yarn/a` and `b` and `c`. Parentheses, though, not brackets :heavy_check_mark: 
```suggestion
    const r = new RegExp(`\\.yarn/(${config.CLEAN_YARN_IGNORES_LIST.join('|')})`)
```